lib/m2ts/metadata-stream.js

```diff
@@ -38,9 +46,46 @@ var
             (data[3]);
   },
   tagParsers = {
-    TXXX: function(tag) {
+    'APIC': function(tag) {
+      var i, j;
+      if (tag.data[0] !== textEncodingDescriptionByte.Utf8) {
+        // ignore frames with unrecognized character encodings
+        return;
+      }
+
+      // parsing fields [ID3v2.4.0 section 4.14.]
+      i = tag.data.indexOf(0, 1);
+      if (i < 0) {
+        return; // malformed frame
+      }
+      
+      tag.mimeType = parseIso88591(tag.data, 1, i++); // MIME type terminated with \0
+      tag.pictureType = tag.data[i++]; // 1-byte Picture Type
+
+      j = tag.data.indexOf(0, i);
+      if (j < 0) {
+        return; // malformed frame
+      }
+
+      tag.description = parseUtf8(tag.data, i, j++); // Description terminated with \0
+      if (tag.mimeType === '-->') {
+        tag.url = parseIso88591(tag.data, j, tag.data.length) // URL (always represented as ISO-8859-1 [ID3v2.4.0 section 4.])
+      } else {
+        tag.pictureData = tag.data.subarray(j, tag.data.length); // Picture data (binary data)
+      }
+    },
+    'T*': function(tag) {
+      if (tag.data[0] !== textEncodingDescriptionByte.Utf8) {
+        // ignore frames with unrecognized character encodings
+        return;
+      }
+      // parse text field, do not include null terminator in the tag value
+      // frames that allow different types of encoding contain terminated text [ID3v2.4.0 section 4.]
+      tag.value = parseUtf8(tag.data, 1, tag.data.length).replace(/\0*$/, '');;
```

Comment:
Yes, but I think it's something new in ID3 v2.4. ID3 v2.3 does not mention that: https://id3.org/id3v2.3.0#Text_information_frames. Also, Wowza ID3 creating Java API does not support multiple strings in TXXX frames.
We should either leave it for the client to handle or handle by mux.js but then I don't know how to avoid breaking backwards compatibility of the `tag.value` field. Should it be either string or array?

Reply:
I've prepared and pushed a change for parsing multiple strings in text information frames. `tag.value` will become string or an array if multiple strings are detected: 556d70864e1794d1d25aab85f6aee7cbab4b46d1

Reply:
Thanks for checking into both options. The idea of returning either a string or an array in this case does seem a bit odd. Since we didn't handle these tags before, we do have an opportunity to decide the best approach.

So we either:
1. Return a string or array in `value`, depending on how many null chars there are
2. Return a string for the client to split if they want
3. Return an array regardless of number
4. Return both `value` with a single value, unsplit, and `values`, which is split on null chars

I'm leaning towards 3, but what do you think @pszemus ? And @gkatsev any thoughts?

Reply:
Hmmm... Regarding option 3: If one uses an encoder that does not support multiple strings in text frames (for example because it uses a 2.3 version of ID3) he/she shouldn't expect an array being returned in `tag.value`.

Reply:
So, right now `tag.value` returns a string and in this PR we're adding parsing a new ID3 types that might have multiple values?
If I'm understanding correctly, I think the best course of action would be 4 as it preserves backwards compatibility without introducing issues of unexpected behavior. `tag.value` will be the only value or the first value if there are multiple items. `tag.values` will be an array of the values. If there's only one value, that array will only have a single item.

Reply:
We're adding new feature here - parsing text frames, that can contain multiple strings. TXXX frames that mux.js supported so far, are left untouched as they are single string frames (accoring to ID3 v2.4 section 4.2.6). So as @gesinger said, we can do almost anything. I'm opting for 1 or 2 (as multiple strings we're introduced in version 2.4 od ID3).

Reply:
What to you think about evolving option 1 to:
`tag.value` - string or array
`tag.hasMultipleValues` - as a boolean indicator whether `tag.value` is string or array 
?

Reply:
Is `tag.value` an existing thing or a new thing? I thought it was an existing thing.

Reply:
I agree with @gkatsev and think we should go with `tag.value` and `tag.values`. Since, as @pszemus pointed out, older versions of ID3, and many encoders, only support one string, so keeping `tag.value` maintains a consistent API and handles the majority of cases. For the case of multiple values, I think it's better not to have the user determine the null char and parse it out themselves, and not worry about the changing type of `value`. We can always include `values` for those who may have multiple strings in a single frame.

Reply:
Agreed. I will make necessary changes next week.

Reply:
Thank you @pszemus !

Reply:
changes pushed, thanks